src/scalaDebugger.ts

```diff
@@ -46,15 +50,161 @@ export async function start(
 
 class ScalaConfigProvider implements vscode.DebugConfigurationProvider {
   provideDebugConfigurations(): ProviderResult<DebugConfiguration[]> {
-    return [];
+    const mainClassPick = "Main Class";
+    const testClassPick = "Test Suite";
+
+    return vscode.window
+      .showQuickPick([mainClassPick, testClassPick], {
+        placeHolder: "Pick kind of class to debug",
+      })
+      .then((result) => {
+        if (result === mainClassPick) {
+          return this.provideDebugMainClassConfiguration().then(
+            (config) => [config],
+            (_) => []
+          );
+        } else if (result === testClassPick) {
+          return this.provideDebugTestClassConfiguration().then(
+            (config) => [config],
+            (_) => []
+          );
+        }
+        return [];
+      });
   }
 
   resolveDebugConfiguration(
     _folder: WorkspaceFolder | undefined,
     debugConfiguration: DebugConfiguration
   ): ProviderResult<DebugConfiguration> {
+    if (debugConfiguration.type === undefined) return null;
     return debugConfiguration;
   }
+
+  private provideDebugMainClassConfiguration(): Thenable<DebugConfiguration> {
+    return this.askForBuildTarget().then((buildTarget) =>
+      this.askForClassName().then((className) =>
+        this.askForArguments().then((args) =>
+          this.askForConfigurationName(className).then((name) => {
+            const result: DebugConfiguration = {
+              type: configurationType,
+              name: name,
+              request: launchRequestType,
+              mainClass: className,
+              buildTarget: buildTarget,
+              args: args,
+            };
+            return result;
+          })
+        )
+      )
+    );
+  }
+
+  private provideDebugTestClassConfiguration(): Thenable<DebugConfiguration> {
+    return this.askForBuildTarget().then((buildTarget) =>
+      this.askForClassName().then((className) =>
+        this.askForConfigurationName(className).then((name) => {
+          const result: DebugConfiguration = {
+            type: configurationType,
+            name: name,
+            request: launchRequestType,
+            testClass: className,
+            buildTarget: buildTarget,
+          };
+          return result;
+        })
+      )
+    );
+  }
+
+  private askForBuildTarget(): Thenable<string | undefined> {
```

Comment:
do we still need `undefined` here? Can't we just use `Promise.reject()` for that case too?

Reply:
yes, `undefined` reflects no `buildTarget` provided.